eth-contracts/contracts/DelegateManager.sol

```diff
@@ -515,10 +527,15 @@ contract DelegateManager is InitializableV2 {
             uint256 newDelegateStake = (
              totalBalanceInStakingAfterSlash.mul(preSlashDelegateStake)
             ).div(totalBalanceInStakingPreSlash);
-            // uint256 slashAmountForDelegator = preSlashDelegateStake.sub(newDelegateStake);
+            // slashAmountForDelegator = preSlashDelegateStake - newDelegateStake;
             delegateInfo[delegator][_slashAddress] = (
                 delegateInfo[delegator][_slashAddress].sub(preSlashDelegateStake.sub(newDelegateStake))
             );
+            // Update total stake for delegator
+            _updateDelegatorTotal(
+                delegator,
+                delegatorTotals[delegator].sub(preSlashDelegateStake.sub(newDelegateStake))
+            );
             // Update total decrease amount
             totalDelegatedStakeDecrease = (
                 totalDelegatedStakeDecrease.add(preSlashDelegateStake.sub(newDelegateStake))
```

Comment:
we now call `preSlashDelegateStake.sub(newDelegateStake))` 3 times, at this point does it make sense to just compute once and store in var?

Reply:
stack too deep issue here, hence recomputation